src/plugins/kibana_utils/docs/state_containers/react/use_selector.md

```diff
@@ -0,0 +1,20 @@
+# `useSelector()` hook
+
+With `useSelector` React hook you specify a selector function, which will pick specific
+data from the state. *Your component will update only when that specific part of the state changes.*
+
+```tsx
+const selector = state => state.isDarkMode;
+const Demo = () => {
+  const isDarkMode = useSelector(selector);
+  return <div>{isDarkMode ? '🌑' : '☀️'}</div>;
+};
+```
+
+As an optional second argument for `useSelector` you can provide a `comparator` function, which
+compares currently selected value with the previous and your component will re-render only if
+`comparator` returns `true`. By default it uses [`fast-deep-equal`](https://github.com/epoberezkin/fast-deep-equal).
```

Comment:
I am not sure it is a good idea to use deep comparison by default. 
I believe comparison overhead could accidently a larger performance hit then not needed render recalculations.
I think in react ecosystem we more used to haveing Object.is by default with possibility to use shallow / deep when need.

Reply:
The idea here is to provide the most convenient option. If `Object.is` would work only for most basic types. But if you chose to use `useSelector` in the first place, you want your component not to re-render when data stays the same, otherwise you might use `useState`.

Reply:
But aren't our transitions enforce changes to be immutable so we could safely use `Object.is`? Just like in redux?

Reply:
Note: discussed offline, my understanding is that we agreed that deep comparison is good here and if need be API allows to easily change it for any other comparator.